### ISSUE TITLE
build error with components folder in pages folder

### ISSUE THREAD

**Issue by gaoljie** [zeit/next.js#9546]
# Bug report

build error with components folder in pages folder

## Describe the bug

I wanted to create custom nav component with home page, so I created a nav component in pages/index/components/nav.js 

nav.js

```js
import React from 'react'

const Nav = ({list}) => (

    <div>
        {list.map(item => <p>item</p>)}
    </div>
)

export default Nav

```

when I try to build the project `next build`, it has this following error

```
Error occurred prerendering page "/index/components/nav": TypeError: Cannot read property 'map' of undefined
    at Nav (/Users/jay/Downloads/my-app/.next/server/static/gvH9SPsJWVGjX-v1v6eEN/pages/index/components/nav.js:116:31)
    at d (/Users/jay/Downloads/my-app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:36:498)
    at Za (/Users/jay/Downloads/my-app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:39:16)
    at a.b.render (/Users/jay/Downloads/my-app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:44:476)
    at a.b.read (/Users/jay/Downloads/my-app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:44:18)
    at renderToString (/Users/jay/Downloads/my-app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:54:364)
    at render (/Users/jay/Downloads/my-app/node_modules/next/dist/next-server/server/render.js:81:16)
    at Object.renderPage (/Users/jay/Downloads/my-app/node_modules/next/dist/next-server/server/render.js:323:20)
    at /Users/jay/Downloads/my-app/.next/server/static/gvH9SPsJWVGjX-v1v6eEN/pages/_document.js:431:21
    at Generator.next (<anonymous>)
```

When I moved components folder out of pages folder, it does not have this issues anymore

## To Reproduce

1. create index folder under pages
2. create components folder under index folder
3. create nav.js file under components folder
```js
import React from 'react'

const Nav = ({list}) => (

    <div>
        {list.map(item => <p>item</p>)}
    </div>
)

export default Nav
```
4. `next build`

**Comment by ijjk** [comment]
Hi, you should not put your components folder in your pages folder. We try to catch and show an error when this is being done since it hinders performance as each JavaScript file in your pages folder is turned into it's own route.

I'm going to close this since we just merged #9548 to help show this when an error occurs during prerendering like you experienced

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
